src/components/transport_manager/test/transport_manager_default_test.cc

```diff
@@ -32,44 +32,70 @@
 #include "gtest/gtest.h"
 #include "transport_manager/transport_manager.h"
 #include "transport_manager/transport_manager_default.h"
-#include "resumption/last_state.h"
 #include "transport_manager/mock_transport_manager_settings.h"
-#include "resumption/last_state_impl.h"
+#include "resumption/mock_last_state.h"
 
 namespace test {
 namespace components {
 namespace transport_manager_test {
 
+using namespace ::resumption;
+using namespace ::Json;
+
+using ::resumption::MockLastState;
 using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::NiceMock;
 
 TEST(TestTransportManagerDefault, Init_LastStateNotUsed) {
   MockTransportManagerSettings transport_manager_settings;
   transport_manager::TransportManagerDefault transport_manager(
       transport_manager_settings);
-  resumption::LastStateImpl last_state("app_storage_folder",
-                                       "app_info_storage2");
+
+  NiceMock<MockLastState> mock_last_state;
+  Value custom_dictionary = Value();
+
+  ON_CALL(mock_last_state, GetDictionary())
+      .WillByDefault(ReturnRef(custom_dictionary));
 
   EXPECT_CALL(transport_manager_settings, use_last_state())
       .WillRepeatedly(Return(false));
   EXPECT_CALL(transport_manager_settings, transport_manager_tcp_adapter_port())
       .WillRepeatedly(Return(1u));
-  transport_manager.Init(last_state);
+
+  transport_manager.Init(mock_last_state);
+  transport_manager.Stop();
 }
 
-TEST(TestTransportManagerDefault, DISABLED_Init_LastStateUsed) {
-  // TODO (dcherniev) : investigate and fix SegFault issue
+TEST(TestTransportManagerDefault, Init_LastStateUsed) {
   MockTransportManagerSettings transport_manager_settings;
   transport_manager::TransportManagerDefault transport_manager(
       transport_manager_settings);
-  resumption::LastStateImpl last_state("app_storage_folder",
-                                       "app_info_storage2");
+
+  NiceMock<MockLastState> mock_last_state;
+  Value custom_dictionary;
+  Value tcp_device;
+  tcp_device["name"] = "unique_tcp_device_name";
+  tcp_device["address"] = "57.48.0.1";
```

Comment:
@LuxoftAKutsan Well actually no, this ip is by no means specific for this test. This ip was taken from LastState data (that was used in previous realization) and I can tell that this data was just copy pasted version of "app_info_storage".